CMakeLists.txt

```diff
@@ -562,17 +553,14 @@ if(WITH_OPTIM)
         add_definitions("-DX86_CPUID")
         set(ZLIB_ARCH_SRCS ${ZLIB_ARCH_SRCS} ${ARCHDIR}/x86.c)
         if(HAVE_SSE42_INTRIN)
-            add_definitions(-DX86_SSE4_2_CRC_HASH)
+            add_definitions(-DX86_SSE4_2)
             set(ZLIB_ARCH_SRCS ${ZLIB_ARCH_SRCS} ${ARCHDIR}/insert_string_sse.c)
-            add_feature_info(SSE4_CRC 1 "Support CRC hash generation using the SSE4.2 instruction set, using \"${SSE4FLAG}\"")
-            add_intrinsics_option("${SSE4FLAG}")
-            if(HAVE_SSE42CRC_INTRIN)
-                add_definitions(-DX86_SSE4_2_CRC_INTRIN)
-            endif()
+            add_feature_info(SSE42 1 "Use the SSE4.2 instruction set, using \"${SSE42FLAG}\"")
+            add_intrinsics_option("${SSE42FLAG}")
             if(WITH_NEW_STRATEGIES)
                 add_definitions(-DX86_QUICK_STRATEGY)
                 set(ZLIB_ARCH_SRCS ${ZLIB_ARCH_SRCS} ${ARCHDIR}/deflate_quick.c)
-                add_feature_info(SSE4DEFLATE 1 "Support SSE4.2-accelerated quick decompression")
+                add_feature_info(SSE42_DEFLATE 1 "Support SSE4.2-accelerated quick decompression")
```

Comment:
I am thinking this should rather be renamed DEFLATE_QUICK for example, since we don't currently have SSE4.2 features in any other deflate strategy, but we might in the future. Deflate-quick is a feature, not an optimization in the existing deflate strategy.

Also, this says decompression, but deflate is compression.